Userland/Libraries/LibAudio/Output.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2023, Gregory Bertilson <zaggy1024@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/AtomicRefCounted.h>
+#include <AK/Queue.h>
+#include <AK/Time.h>
+#include <LibAudio/OutputImplementation.h>
+#include <LibAudio/SampleFormats.h>
+#include <LibThreading/ConditionVariable.h>
+#include <LibThreading/MutexProtected.h>
+#include <LibThreading/Thread.h>
+
+namespace Audio {
+
+enum class OutputState {
+    Playing,
+    Suspended,
+};
+
+class Output : public AtomicRefCounted<Output> {
```

Comment:
To be clear: This is a Ladybird cross-platform-compatibility feature, not an attempt at generalizing our LibAudio output functionality for cross-platform use. I don't want to support a whole host of audio backends through a generic interface, I want to keep LibAudio's main output functionality away from anything cross-platform, though we can plug in ConnectionToAudioServer into a backend for this Ladybird interface. Maybe specify that in the comment; `Output` is too generic for my taste.